distributed/utils.py

```diff
@@ -1041,7 +1063,8 @@ def import_file(path):
             import pkg_resources
             pkgs = pkg_resources.find_distributions(path)
             for pkg in pkgs:
-                names_to_import.append(pkg.project_name)
+                # See https://github.com/dask/distributed/issues/1819
+                names_to_import.extend(pkg.get_metadata_lines('top_level.txt'))
```

Comment:
cc @jakirkham, who probably knows more about packaging than I do.

Reply:
@jakirkham if you find yourself with a few free minutes can I ask you to look this over briefly?

Reply:
I spoke with @jakirkham briefly and he didn't have a quick answer.  I think he'll take a longer look sometime today or tomorrow.

Reply:
Thoughts on using [`pkgutil.iter_modules`]( https://docs.python.org/3/library/pkgutil.html#pkgutil.iter_modules )?

Reply:
`pkgutil.iter_modules` is better it is more general: it applies to the zip and pyz case as well, and it works for custom importers. I've switched the code over to that

Reply:
Glad to hear that works. Also nice to see the code simplification here 🙂